File: 70-climbing-stairs/70-climbing-stairs.go
func climbStairs(n int) int {
    if n == 1 {
        return 1
    }
    if n == 2 {
        return 2
    }
    ways := make([]int, n)
    ways[0] = 1
    ways[1] = 2
    for i := 2; i < n; i++ {
        ways[i] = ways[i-1] + ways[i-2]
    }
    
    return ways[n-1]
}

File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        if n == 2:
            return 2
        ways = [1,2]
        for i in range(n-2):
            ways.append(ways[i] + ways[i+1])
        return ways[-1]
        